src/widgets/ot_carousel.eliom

```diff
@@ -426,10 +427,22 @@ let%shared bullets
            :: R.a_class class_
            :: a_onclick [%client fun _ -> ~%change (`Goto ~%i)]
            :: a)
-      []
+      c
+  in
+  let rec aux acc i cl =
+    if i = 0
+    then acc
+    else
+      let acc, cl = match cl with
+	| c::cl ->
+	  bullet (i-1) c :: acc, cl
+	| _ ->
+	  bullet (i-1) [] :: acc, []
+      in
+      aux acc (i-1) cl
```

Comment:
I think you should fail (by `invalid_arg`) if `length <> List.length content`. Filling in with empty bullets (or ignoring the tail of `content`) looks arbitrary. Assuming matching lengths, you can use `List.mapi` instead of `aux`.